main.tf

```
@@ -5,8 +5,8 @@ locals {
   root_iops            = var.root_volume_type == "io1" ? var.root_iops : "0"
   ebs_iops             = var.ebs_volume_type == "io1" ? var.ebs_iops : "0"
   availability_zone    = var.availability_zone != "" ? var.availability_zone : data.aws_subnet.default.availability_zone
-  ami                  = var.ami != "" ? var.ami : data.aws_ami.default.image_id
-  ami_owner            = var.ami != "" ? var.ami_owner : data.aws_ami.default.owner_id
+  ami                  = var.ami != "" ? var.ami : data.aws_ami.default.0.image_id
+  ami_owner            = var.ami != "" ? var.ami_owner : data.aws_ami.default.0.owner_id
```

```suggestion
  ami                  = var.ami != "" ? var.ami : join("", data.aws_ami.default.*.image_id)
  ami_owner            = var.ami != "" ? var.ami_owner : join("", data.aws_ami.default.*.owner_id)
```
use join+splash instead of index lookup, please